### PR TITLE
PT-903 Patch platform-data to prevent from Android bug to happen

### PR DESCRIPTION
**Pull request by franck-fourel**

### CODE REVIEW

**Comment by franck-fourel** [comment]
I don't like what I did but at least it works and the tests are working. 
The thing that is not convenient is the fact that we do not log the warnings in the logs in when we have a warning and skip the related data.

File: data/test/datum.go [hunk 1]
```diff
@@ -80,6 +80,12 @@ func (d *Datum) Validate(validator structure.Validator) {
 	d.ValidateInputs = append(d.ValidateInputs, validator)
 }
 
+// IsValid
+// returns true if there is no error and no warning in the validator
```
Comment (yvon-dblg) [inline data/test/datum.go]: `// IsValid returns true if there is no error and no warning in the validator`
golint: ./data/test/datum.go:83:1: comment on exported method Datum.IsValid should be of the form "IsValid ..."

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,10 @@
 # Platform 
 The Tidepool platform API
 
+## unreleased
+### Fixed
+- PT-9.3 Patch platform-data to prevent from Android bug to happen
```
Comment (yvon-dblg) [inline CHANGELOG.md]:
```suggestion
- PT-903 Patch platform-data to prevent an Android bug to happen
```

**Review by yvon-dblg — CHANGES_REQUESTED** [review]
golint warnings:
```
golint
17a18,30
> ./data/test/datum.go:83:1: comment on exported method Datum.IsValid should be of the form "IsValid ..."
> ./data/types/activity/physical/physical.go:270:1: comment on exported method Physical.IsValid should be of the form "IsValid ..."
> ./data/types/basal/automated/automated.go:76:1: comment on exported method Automated.IsValid should be of the form "IsValid ..."
> ./data/types/basal/scheduled/scheduled.go:76:1: comment on exported method Scheduled.IsValid should be of the form "IsValid ..."
> ./data/types/basal/suspend/suspend.go:73:1: comment on exported method Suspend.IsValid should be of the form "IsValid ..."
> ./data/types/basal/temporary/temporary.go:89:1: comment on exported method Temporary.IsValid should be of the form "IsValid ..."
> ./data/types/base.go:227:1: comment on exported method Base.IsValid should be of the form "IsValid ..."
> ./data/types/blood/glucose/glucose.go:27:1: comment on exported method Glucose.IsValid should be of the form "IsValid ..."
> ./data/types/blood/ketone/ketone.go:47:1: comment on exported method Ketone.IsValid should be of the form "IsValid ..."
> ./data/types/bolus/combination/combination.go:113:1: comment on exported method Combination.IsValid should be of the form "IsValid ..."
> ./data/types/bolus/extended/extended.go:81:1: comment on exported method Extended.IsValid should be of the form "IsValid ..."
> ./data/types/bolus/normal/normal.go:65:1: comment on exported method Normal.IsValid should be of the form "IsValid ..."
> ./data/types/calculator/calculator.go:105:1: comment on exported method Calculator.IsValid should be of the form "IsValid ..."
18a32,45
> ./data/types/device/alarm/alarm.go:90:1: comment on exported method Alarm.IsValid should be of the form "IsValid ..."
> ./data/types/device/calibration/calibration.go:53:1: comment on exported method Calibration.IsValid should be of the form "IsValid ..."
> ./data/types/device/prime/prime.go:74:1: comment on exported method Prime.IsValid should be of the form "IsValid ..."
> ./data/types/device/reservoirchange/reservoirchange.go:62:1: comment on exported method ReservoirChange.IsValid should be of the form "IsValid ..."
> ./data/types/device/status/status.go:116:1: comment on exported method Status.IsValid should be of the form "IsValid ..."
> ./data/types/device/timechange/timechange.go:97:1: comment on exported method TimeChange.IsValid should be of the form "IsValid ..."
> ./data/types/food/food.go:102:1: comment on exported method Food.IsValid should be of the form "IsValid ..."
> ./data/types/insulin/insulin.go:61:1: comment on exported method Insulin.IsValid should be of the form "IsValid ..."
> ./data/types/settings/cgm/cgm.go:111:1: comment on exported method CGM.IsValid should be of the form "IsValid ..."
> ./data/types/settings/pump/pump.go:151:1: comment on exported method Pump.IsValid should be of the form "IsValid ..."
> ./data/types/state/reported/reported.go:51:1: comment on exported method Reported.IsValid should be of the form "IsValid ..."
> ./data/types/test/test.go:85:1: comment on exported function ValidateWarningsWithOrigin should be of the form "ValidateWarningsWithOrigin ..."
> ./data/types/test/test.go:93:1: comment on exported function ValidateWarningsWithExpectedOrigins should be of the form "ValidateWarningsWithExpectedOrigins ..."
> ./data/types/upload/upload.go:173:1: comment on exported method Upload.IsValid should be of the form "IsValid ..."
```

Another approach may have been to convert any negative value to the positive one:
duration := uint32(duration)
But I'm not sure if there will be less code to do.



Looks good, good job.

**Review by yvon-dblg — APPROVED** [review]

**Pull request merged**